.github/workflows/pytest.yml

```diff
@@ -15,12 +15,25 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  pytest:
+  select-strategy:
     runs-on: ubuntu-latest
+    outputs:
+      python-versions: ${{ steps.set-matrix.outputs.python-versions }}
+    steps:
+      - id: set-matrix
+        run: |
+          if [ ${{ github.event_name }} == "release" ]; then
```

Comment:
If we run this only for PRs towards `master` (which is basically a release) the logic might be bit simpler no?

Reply:
I'm not sure I follow, do you mean the `select-strategy` job should run only on PR that target `master`? I think this becomes a problem in the `pytest` job because of the `needs: select-strategy` line, i.e. the way it is written it is expected `select-strategy` is run in all cases.

I've tried to implement this using only "workflow logic" but couldn't really come up with a good solution, so that's why I had to incorporate these bash lines instead.